server/src/main/java/io/seata/server/coordinator/DefaultCoordinator.java

```diff
@@ -331,6 +332,8 @@ protected void timeoutCheck() {
 
                 // transaction timeout and start rollbacking event
                 MetricsPublisher.postSessionDoingEvent(globalSession, GlobalStatus.TimeoutRollbacking.name(), false, false);
+                // do rollback
+                core.doGlobalRollback(globalSession, false);
```

Comment:
`timeoutCheck` as mark thread, don't add time-consuming operations to it.

Reply:
I have moved the operation to 'retryRollbacking' thread and change the status to 'Rollbacking' after the first successful rollback